### DISCUSSION TITLE
NumericalFeatureTokenizer after train test split

### DISCUSSION THREAD

**Discussion by OrtalDayan** [yandex-research/rtdl#45] [Q&A]
Looking at the code it seems to me that it doesn't matter (won't cause data leakage ) if we apply the NumericalFeatureTokenizer or NumEmbeddings class before or after train test split. Am I right?

**Comment by Yura52** [comment]
These are PyTorch modules in the same way as, for example, `torch.nn.Linear` is. It means that:
- they are end-to-end trainable parts of a model, not preprocessing tools
- they are always "applied" to the data, not just once (as with preprocessing tools such as `sklearn.preprocessing.StandardScaler`)

Does this help?

**Reply by OrtalDayan** [reply]
Yes. Thank you very much.
